src/components/Netcat/Netcat.tsx

```diff
@@ -30,19 +21,46 @@ interface FormValuesType {
+    //Component State Variables
+    var [output, setOutput] = useState(""); // State variable to store the output of the command execution.
     const [selectedScanOption, setSelectedNetcatOption] = useState("");
-    const [pid, setPid] = useState("");
-    const [loading, setLoading] = useState(false);
-    const [allowSave, setAllowSave] = useState(false);
-    const [hasSaved, setHasSaved] = useState(false);
-    const [checkedVerboseMode, setCheckedVerboseMode] = useState(false);
-
+    const [pid, setPid] = useState(""); // State variable to store the process ID of the command execution.
+    const [isCommandAvailable, setIsCommandAvailable] = useState(false); // State variable to check if the command is available.
+    const [loading, setLoading] = useState(false); // State variable to indicate loading state.
+    const [allowSave, setAllowSave] = useState(false); // State variable to allow saving the output to a file
+    const [hasSaved, setHasSaved] = useState(false); // State variable to indicate if the output has been saved
+    const [checkedVerboseMode, setCheckedVerboseMode] = useState(false); // State variable to indicate whether the verbose mode is enabled.
+    const [loadingModal, setLoadingModal] = useState(true); // State variable to indicate loading state of the modal.
+    const [opened, setOpened] = useState(!isCommandAvailable); // State variable that indicates if the modal is opened.
+    // Component Constants.
+    const title = "Netcat Tool"; // Title of the component.
+    const description =
+        "Netcat is a powerful tool that is used to connect two machines together for communication and for other uses."; // Description of the component.
+    const steps =
+        "How to use this netcat tool:\n" +
+        "- If you want to listen for connections for chat or reverse shell choose the Interactive shell/Listen option and provide \n a port number.\n" +
+        "- If you want to scan for ports, provide an IP address and a port range\n" +
+        "- If you want to send a file, provide the destination IP address, Port number, and File name\n" +
+        "- If you want to receive a file, provide a port number and the File name.\n" +
+        "- If you want to port scan a domain, provide Domain name and a Port number. \n" +
+        "Note: You should only use website port scan to a domain that you own.\n" +
+        "Note 2: Using the sending/receiving file option might seem like it is not working, but it is working.\n" +
+        "You will need a second machine to see the file transfer\n" +
+        "For more information go to the reference page and click on netcat, alternatively Google is your friend.";
+    const sourceLink = "https://github.com/openbsd/src/blob/master/usr.bin/nc/netcat.c"; // Link to the source code
+    const tutorial = ""; // Link to the official documentation/tutorial.
+    const dependencies = ["netcat"]; // Contains the dependencies required by the component
+
```

Comment:
useEffect component should go here to check for dependencies. 
```suggestion

 // Check if the command is available and set the state variables accordingly.
    useEffect(() => {
        // Check if the command is available and set the state variables accordingly.
        checkAllCommandsAvailability(dependencies)
            .then((isAvailable) => {
                setIsCommandAvailable(isAvailable); // Set the command availability state
                setOpened(!isAvailable); // Set the modal state to opened if the command is not available
                setLoadingModal(false); // Set loading to false after the check is done
            })
            .catch((error) => {
                console.error("An error occurred:", error);
                setLoadingModal(false); // Also set loading to false in case of error
            });
    }, []);
```